zerver/lib/test_runner.py

```diff
@@ -134,7 +135,22 @@ class Process(multiprocessing.context.ForkProcess):
 def destroy_test_databases(worker_id: Optional[int]=None) -> None:
     for alias in connections:
         connection = connections[alias]
-        try:
+
+        def monkey_patched_destroy_test_db(test_database_name: str, verbosity: Any) -> None:
+            """
+            We need to monkey-patch connection.creation._destroy_test_db to
+            use the IF EXISTS parameter - we don't have a guarantee that the
+            database we're cleaning up actually exists and since Django 3.1 the original implementation
+            throws an ugly `RuntimeError: generator didn't stop after throw()` exception and triggers
+            a confusing warnings.warn inside the postgresql backend implementation in _nodb_cursor()
+            if the database doesn't exist.
```

Comment:
We should file an issue with Django upstream for having an explicit way to configure "database doesn't exist is OK", I guess?

Reply:
https://code.djangoproject.com/ticket/32376 - added this link to the docstring of the monkey patched funcion as well